fix(preview-middleware): default flp.libs parameter to true

heimwege opened the pull request:
- default the `flp.libs` parameter of `preview-middleware` to `true` for edmx backends
- override the `flp.libs` parameter of `preview-middleware` to `false` for cap
- for compatibility reasons: enhance the `serve-static-middleware` to additionally serve paths that start with `/resources` w/o the `/resources` part (excluding `/resources` as standalone path) to ensure that the workaround to use servestatic on `/resources/my/custom/lib` because the `flp.libs` parameter was not known/used will still continue to work.

Sample: An existing `path` configuration `/resources/my/custom/lib` to serve a custom library locally w/o `flp.libs` `true` will have two servestatic paths.
1. `/resources/my/custom/lib` <-- the existing one remains unchanged, but will no longer be used because
2. `/my/custom/lib          ` <-- the new `flp.libs` default `true` will add the custom library and its respective ABAP backend url to the  UI5 loader configuration; That's why we add this `path` configuration on-the-fly to ensure compatibility with the new `flp.libs` `true` default

Sample: An existing `path` configuration `/resources` will remain unchanged and still only has one servestatic path so that the whole UI5 framework can still be served locally
1. `/resources` <-- the existing one remains unchanged, and will continue to work

heimwege committed:
default flp.libs to true and enhance servestatic

heimwege committed:
ignore serve static paths that match '/resources'

heimwege committed:
add unit test

heimwege committed:
adjust snaps

heimwege committed:
fix lint errors

github-actions[bot] committed:
Linting auto fix commit

heimwege committed:
use manifest type from @sap-ux/project-access instead of defining own one

heimwege committed:
Merge remote-tracking branch 'origin/fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
fix sonar errors

heimwege committed:
remove todo

github-actions[bot] committed:
Linting auto fix commit

heimwege committed:
refactoring

heimwege committed:
Merge remote-tracking branch 'origin/fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
refactoring (part 2)

heimwege committed:
fix log message only when flp.libs = true for non edmx projects

heimwege committed:
add fallback libs for offline use case

github-actions[bot] committed:
Linting auto fix commit

heimwege committed:
add changesets

heimwege committed:
Merge remote-tracking branch 'origin/fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
fix + add unit test

heimwege committed:
fix sonar issue

heimwege committed:
refactoring

heimwege committed:
fix sonar issues

heimwege committed:
add unit tests for error cases

heimwege committed:
harmonize usage of UI5 version info types

heimwege committed:
one day I'll find out how to share types between preview-middleware-client and preview-middleware

heimwege committed:
disable @typescript-eslint/no-duplicate-type-constituents

heimwege committed:
disable @typescript-eslint/no-duplicate-type-constituents

heimwege committed:
fix lint errors

github-actions[bot] committed:
Linting auto fix commit

heimwege committed:
fix lint error

heimwege committed:
fix lint error

heimwege committed:
Merge branch 'fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
fix lint error

heimwege commented on packages/preview-middleware-client/src/utils/version.ts:
```diff
@@ -2,12 +2,30 @@ import VersionInfo from 'sap/ui/VersionInfo';
+     * - the id of the application in case the UI5 sources have beeng loaded from npmjs.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-duplicate-type-constituents, @typescript-eslint/no-redundant-type-constituents
+    name: 'SAPUI5 Distribution' | Manifest['sap.app']['id'];
```
Actually this is `string | sting` which makes no sense but it was done like this to point out where the second string value actually comes from. This helps when using code completion.

heimwege commented on packages/preview-middleware/src/base/flp.ts:
```diff
@@ -60,6 +60,38 @@ import { readFileSync } from 'node:fs';
+     * - the name of the distribution or
+     * - the id of the application in case the UI5 sources have beeng loaded from npmjs.
+     */
+    name: 'SAPUI5 Distribution' | Manifest['sap.app']['id'];
```
Actually this is `string | sting` which makes no sense but it was done like this to point out where the second string value actually comes from. This helps when using code completion.

I'm wondering why eslint does not complain about the `string | string` issue here. This is only checked in `preview-middleware-client` `version.ts` 🤷🏻

heimwege commented on packages/preview-middleware/src/base/flp.ts:
```diff
@@ -60,6 +60,38 @@ import { readFileSync } from 'node:fs';
+     * - the version of the UI5 framework or
+     * - the version of the application in case the UI5 sources have been loaded from npmjs.
+     */
+    version: string | Manifest['sap.app']['applicationVersion']['version'];
```
Actually this is `string | sting` which makes no sense but it was done like this to point out where the second string value actually comes from. This helps when using code completion.

I'm wondering why eslint does not complain about the `string | string` issue here. This is only checked in `preview-middleware-client` `version.ts` 🤷🏻

tobiasqueck reviewed (dismissed):
The PR contains quite a lot of changes but in the end you just set the default to `true` for `edmx` and added a workaround to the serve static middleware. The rest is code restructurings and cleanups, correct?
That mix made the review a little more time consuming but in the end, the code is now cleaner and better structured. The functional change is also a good idea and should have been done a long time ago (I missed it when creating the initial version of this code). The workaround in the serve static middleware will cause confusion in the future - I am sure of it - but I understand it is required to keep existing setups working as-is.

devinea reviewed (dismissed):
Thanks @heimwege. Changes look good.
Haven't tested locally.
changeset ✅

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware/README.md

heimwege committed:
rephrase backward compatibility comment in servestatic

heimwege committed:
use type LibraryInfo from sap.ui.core

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
add unit test

github-actions[bot] committed:
Linting auto fix commit

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

longieirl committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware/README.md

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware/src/base/flp.ts

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

longieirl reviewed (dismissed):
Changesets are good.
Changes are clear.
Tests to cover change.

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

github-actions[bot] committed:
Linting auto fix commit

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware/src/base/flp.ts
#	packages/preview-middleware/test/unit/base/__snapshots__/flp.test.ts.snap

heimwege committed:
Merge remote-tracking branch 'origin/fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
AI review comments + adjust test snaps

heimwege committed:
fix missing newline

devinea reviewed (dismissed):
New approach seems light a better way to check for required libs and where to load them from. 
Code changes and tests lokk good. 
I haven't tested locally. 
Changeset 👍

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware/src/base/flp.ts

devinea reviewed (dismissed):
Re-approving

ytpo-lyne committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

ytpo-lyne committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware-client/src/flp/init.ts
#	packages/preview-middleware-client/test/unit/flp/init.test.ts

devinea committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

devinea committed:
Resolve Windows CI build failures and nx build ordering

devinea committed:
fix tests

devinea committed:
ci: switch Playwright tests from chrome to chromium

Google's Chrome dmg URL now returns a 302 redirect that Playwright's
reinstall_chrome_stable_mac.sh does not follow, breaking macOS CI.
Use chromium channel instead and bump cache keys to invalidate old
Chrome-containing caches.

devinea committed:
fix(preview-middleware): use double quotes in build:copy-client for Windows

Single quotes are not interpreted as quoting by Windows cmd.exe, causing
the copyfiles glob to be passed as literal text and resulting in
dist/client being empty. Switch to escaped double quotes which work on
all platforms.

devinea committed:
chore(ci): bump Playwright to 1.60.0 and fix browser cache path

- Upgrade @playwright/test 1.58.2 → 1.60.0 (root, adaptation-editor-tests,
  @sap-ux-private/playwright). 1.58.2's chromium install hangs after
  download on macOS arm64 + Node 24.
- Fix malformed actions/cache `path` for Playwright browsers: the inline
  ${{ runner.os == 'Windows' }}: conditional was being passed as literal
  text. Replace with a workspace-relative cache directory plus
  PLAYWRIGHT_BROWSERS_PATH so cache and install agree on every OS.

devinea committed:
Merge remote-tracking branch 'origin/main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

* origin/main:
  ci: fix Playwright install on macOS (switch to chromium, bump to 1.60.0) (#4773)

devinea committed:
update lock

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

devinea committed:
Merge remote-tracking branch 'origin/main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

* origin/main:
  fix(fiori-mcp-server): exclude Windows GPU DLLs to reduce tgz below npm publish limit (#4783)
  chore: apply latest changesets
  fix(fiori-mcp-server): download ONNX model at runtime to reduce tgz below npm 100 MB limit (#4782)
  chore: apply latest changesets
  bundle mcp server and replace @xenova/transformers with @huggingface/transformers (#4663)
  chore: apply latest changesets
  Revert generator-odata-downloader to cjs (#4775)
  chore: apply latest changesets

devinea committed:
ci: align Playwright cache path with PLAYWRIGHT_BROWSERS_PATH

Cache, install, and test steps were targeting different directories,
so installs were skipped on cache hits but tests then failed to find
the browser executable. Use ${{ github.workspace }}/.ms-playwright
consistently across all three steps in pipeline.yml and
integration-tests.yml.

heimwege committed:
Merge branch 'main' of github.com:SAP/open-ux-tools into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends

# Conflicts:
#	packages/preview-middleware/README.md

heimwege committed:
delete superfluous changeset

heimwege committed:
fix wrong type

heimwege committed:
update snaps

heimwege committed:
undo .github/* changes

heimwege committed:
undo some changes to lock file

heimwege committed:
undo some changes to package.json

heimwege committed:
Merge branch 'main' into fix/preview-middleware/default-flp.libs-to-true-for-edmx-backends
